packages/open-payments/src/types.ts

```diff
@@ -1,9 +1,44 @@
-import { components, paths as Paths } from './generated/types'
+import {
+  components as RSComponents,
+  paths as RSPaths
+} from './generated/resource-server-types'
+import {
+  components as ASComponents,
+  paths as ASPaths,
+  operations as ASOperations
+} from './generated/auth-server-types'
 
-export const getPath = <P extends keyof Paths>(path: P): string =>
+export const getRSPath = <P extends keyof RSPaths>(path: P): string =>
   path as string
-
 export type IncomingPayment =
-  components['schemas']['incoming-payment-with-connection']
-export type ILPStreamConnection = components['schemas']['ilp-stream-connection']
-export type PaymentPointer = components['schemas']['payment-pointer']
+  RSComponents['schemas']['incoming-payment-with-connection']
+export type ILPStreamConnection =
+  RSComponents['schemas']['ilp-stream-connection']
+export type PaymentPointer = RSComponents['schemas']['payment-pointer']
+
+export const getASPath = <P extends keyof ASPaths>(path: P): string =>
+  path as string
+export type NonInteractiveGrantRequest = {
+  access_token: ASOperations['post-request']['requestBody']['content']['application/json']['access_token']
+  client: ASOperations['post-request']['requestBody']['content']['application/json']['client']
+}
+export type NonInteractiveGrant = {
+  access_token: ASComponents['schemas']['access_token']
+  continue: ASComponents['schemas']['continue']
+}
+export type GrantRequest = {
+  access_token: ASOperations['post-request']['requestBody']['content']['application/json']['access_token']
+  client: ASOperations['post-request']['requestBody']['content']['application/json']['client']
+  interact?: ASOperations['post-request']['requestBody']['content']['application/json']['interact']
```

Comment:
@wilsonianb single grant request with optional `interact?` in line with the Open Payments API

Reply:
I was thinking we'd require `interact``
https://github.com/interledger/rafiki/pull/727#discussion_r1025369118
https://github.com/interledger/rafiki/pull/727#discussion_r1025384369
but I'm ok as is. The grant request would just throw if interaction is required and `interact` wasn't specified.

Reply:
Was going back and forth but less errors and throws are better here IMO. Made it required.